src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsdocumentmanagementapi/resource/DocumentController.kt

```diff
@@ -190,10 +191,80 @@ class DocumentController(
       description = "The metadata describing the uploaded document. Should contain a person identifier e.g. prison number or DELIUS id along with any other pertinent metadata. " +
         "The document type used will specify what metadata is required as a minimum",
       required = true,
+      example =
+      """
+      {
+        "prisonCode": "KMI",
+        "prisonNumber": "C3456DE",
+        "court": "Birmingham Magistrates",
+        "warrantDate": "2023-11-14"
+      }
+      """,
     )
     metadata: String,
   ) = documentService.uploadDocument(documentType, documentUuid, file, JacksonUtil.toJsonNode(metadata))
 
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  @PutMapping("/{documentUuid}/metadata")
+  @Operation(
+    summary = "Replace the metadata associated with a document",
+    description = "Accepts JSON based metadata to associate with the document identified by the supplied unique identifier. " +
+      "Applies authorisation and validation rules based on the type of document. If valid, the previous metadata will be stored " +
+      "and the metadata associated with the document will be replaced with the supplied metadata.",
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "202",
+        description = "Document metadata replaced successfully",
+        content = [Content(schema = Schema(implementation = Document::class))],
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role. Note that the required role can be document type dependent",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @PreAuthorize("hasAnyRole('DOCUMENT_WRITER', 'DOCUMENT_ADMIN')")
+  fun replaceDocumentMetadata(
+    @PathVariable
+    @Parameter(
+      description = "Document unique identifier",
+      required = true,
+    )
+    documentUuid: UUID,
+    @RequestBody
+    @Parameter(
+      description = "The replacement metadata describing the document. Should contain a person identifier e.g. " +
+        "prison number or DELIUS id along with any other pertinent metadata. " +
```

Comment:
The person identifier in Delius is case reference number or CRN FYI rather than DELIUS id